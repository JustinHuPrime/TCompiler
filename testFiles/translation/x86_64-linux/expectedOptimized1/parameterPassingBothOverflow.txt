testFiles/translation/x86_64-linux/input/parameterPassingBothOverflow.tc:
TEXT(GLOBAL(_T3foo3bar),
  BLOCK(5,
    MOVE(TEMP(temp6, 1, 1, GP), REG(rdi, 1)),
    MOVE(TEMP(temp7, 2, 2, GP), REG(rsi, 2)),
    MOVE(TEMP(temp8, 4, 4, GP), REG(rdx, 4)),
    MOVE(TEMP(temp9, 8, 8, GP), REG(rcx, 8)),
    MOVE(TEMP(temp10, 1, 1, GP), REG(r8, 1)),
    MOVE(TEMP(temp11, 1, 1, GP), REG(r9, 1)),
    STK_LOAD(TEMP(temp12, 2, 2, GP), CONSTANT(8, LONG(8))),
    MOVE(TEMP(temp13, 4, 4, FP), REG(xmm0, 4)),
    MOVE(TEMP(temp14, 8, 8, FP), REG(xmm1, 8)),
    MOVE(TEMP(temp15, 4, 4, FP), REG(xmm2, 4)),
    MOVE(TEMP(temp16, 4, 4, FP), REG(xmm3, 4)),
    MOVE(TEMP(temp17, 8, 8, FP), REG(xmm4, 8)),
    MOVE(TEMP(temp18, 4, 4, FP), REG(xmm5, 4)),
    MOVE(TEMP(temp19, 8, 8, FP), REG(xmm6, 8)),
    MOVE(TEMP(temp20, 8, 8, FP), REG(xmm7, 8)),
    STK_LOAD(TEMP(temp21, 4, 4, FP), CONSTANT(8, LONG(16))),
    JUMP(LOCAL(3)),
  ),
  BLOCK(3,
    MOVE(TEMP(temp24, 8, 8, GP), GLOBAL(_T3foo3bar)),
    JUMP(LOCAL(22)),
  ),
  BLOCK(22,
    MOVE(TEMP(temp26, 1, 1, GP), TEMP(temp6, 1, 1, GP)),
    JUMP(LOCAL(25)),
  ),
  BLOCK(25,
    MOVE(TEMP(temp28, 2, 2, GP), TEMP(temp7, 2, 2, GP)),
    JUMP(LOCAL(27)),
  ),
  BLOCK(27,
    MOVE(TEMP(temp30, 4, 4, GP), TEMP(temp8, 4, 4, GP)),
    JUMP(LOCAL(29)),
  ),
  BLOCK(29,
    MOVE(TEMP(temp32, 8, 8, GP), TEMP(temp9, 8, 8, GP)),
    JUMP(LOCAL(31)),
  ),
  BLOCK(31,
    MOVE(TEMP(temp34, 1, 1, GP), TEMP(temp10, 1, 1, GP)),
    JUMP(LOCAL(33)),
  ),
  BLOCK(33,
    MOVE(TEMP(temp36, 1, 1, GP), TEMP(temp11, 1, 1, GP)),
    JUMP(LOCAL(35)),
  ),
  BLOCK(35,
    MOVE(TEMP(temp38, 2, 2, GP), TEMP(temp12, 2, 2, GP)),
    JUMP(LOCAL(37)),
  ),
  BLOCK(37,
    MOVE(TEMP(temp40, 4, 4, FP), TEMP(temp13, 4, 4, FP)),
    JUMP(LOCAL(39)),
  ),
  BLOCK(39,
    MOVE(TEMP(temp42, 8, 8, FP), TEMP(temp14, 8, 8, FP)),
    JUMP(LOCAL(41)),
  ),
  BLOCK(41,
    MOVE(TEMP(temp44, 4, 4, FP), TEMP(temp15, 4, 4, FP)),
    JUMP(LOCAL(43)),
  ),
  BLOCK(43,
    MOVE(TEMP(temp46, 4, 4, FP), TEMP(temp16, 4, 4, FP)),
    JUMP(LOCAL(45)),
  ),
  BLOCK(45,
    MOVE(TEMP(temp48, 8, 8, FP), TEMP(temp17, 8, 8, FP)),
    JUMP(LOCAL(47)),
  ),
  BLOCK(47,
    MOVE(TEMP(temp50, 4, 4, FP), TEMP(temp18, 4, 4, FP)),
    JUMP(LOCAL(49)),
  ),
  BLOCK(49,
    MOVE(TEMP(temp52, 8, 8, FP), TEMP(temp19, 8, 8, FP)),
    JUMP(LOCAL(51)),
  ),
  BLOCK(51,
    MOVE(TEMP(temp54, 8, 8, FP), TEMP(temp20, 8, 8, FP)),
    JUMP(LOCAL(53)),
  ),
  BLOCK(53,
    MOVE(TEMP(temp55, 4, 4, FP), TEMP(temp21, 4, 4, FP)),
    JUMP(LOCAL(23)),
  ),
  BLOCK(23,
    SUB(REG(rsp, 8), REG(rsp, 8), CONSTANT(8, LONG(16))),
    MOVE(REG(rdi, 1), TEMP(temp26, 1, 1, GP)),
    MOVE(REG(rsi, 2), TEMP(temp28, 2, 2, GP)),
    MOVE(REG(rdx, 4), TEMP(temp30, 4, 4, GP)),
    MOVE(REG(rcx, 8), TEMP(temp32, 8, 8, GP)),
    MOVE(REG(r8, 1), TEMP(temp34, 1, 1, GP)),
    MOVE(REG(r9, 1), TEMP(temp36, 1, 1, GP)),
    STK_STORE(CONSTANT(8, LONG(0)), TEMP(temp38, 2, 2, GP)),
    MOVE(REG(xmm0, 4), TEMP(temp40, 4, 4, FP)),
    MOVE(REG(xmm1, 8), TEMP(temp42, 8, 8, FP)),
    MOVE(REG(xmm2, 4), TEMP(temp44, 4, 4, FP)),
    MOVE(REG(xmm3, 4), TEMP(temp46, 4, 4, FP)),
    MOVE(REG(xmm4, 8), TEMP(temp48, 8, 8, FP)),
    MOVE(REG(xmm5, 4), TEMP(temp50, 4, 4, FP)),
    MOVE(REG(xmm6, 8), TEMP(temp52, 8, 8, FP)),
    MOVE(REG(xmm7, 8), TEMP(temp54, 8, 8, FP)),
    STK_STORE(CONSTANT(8, LONG(8)), TEMP(temp55, 4, 4, FP)),
    CALL(TEMP(temp24, 8, 8, GP)),
    ADD(REG(rsp, 8), REG(rsp, 8), CONSTANT(8, LONG(16))),
    RETURN(),
  ),
)
