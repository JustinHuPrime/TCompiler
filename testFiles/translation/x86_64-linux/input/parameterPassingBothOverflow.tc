module foo;

void bar(ubyte a, short b, uint c, long d, char e, bool f,
         short inMemory1,
         float fa, double fb, float fc, float fd,
         double fe, float ff, double fg, double fh,
         float inMemory2) {}