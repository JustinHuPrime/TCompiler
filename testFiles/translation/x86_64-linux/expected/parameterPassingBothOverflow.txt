testFiles/translation/x86_64-linux/input/parameterPassingBothOverflow.tc:
TEXT(_T3foo3bar,
  BLOCK(3,
    MOVE(1, REG(rdi), TEMP(temp4, 1, 1, GP)),
    MOVE(2, REG(rsi), TEMP(temp5, 2, 2, GP)),
    MOVE(4, REG(rdx), TEMP(temp6, 4, 4, GP)),
    MOVE(8, REG(rcx), TEMP(temp7, 8, 8, GP)),
    MOVE(1, REG(r8), TEMP(temp8, 1, 1, GP)),
    MOVE(1, REG(r9), TEMP(temp9, 1, 1, GP)),
    STK_LOAD(2, TEMP(temp10, 2, 2, GP), OFFSET(8)),
    MOVE(4, REG(xmm0), TEMP(temp11, 4, 4, FP)),
    MOVE(8, REG(xmm1), TEMP(temp12, 8, 8, FP)),
    MOVE(4, REG(xmm2), TEMP(temp13, 4, 4, FP)),
    MOVE(4, REG(xmm3), TEMP(temp14, 4, 4, FP)),
    MOVE(8, REG(xmm4), TEMP(temp15, 8, 8, FP)),
    MOVE(4, REG(xmm5), TEMP(temp16, 4, 4, FP)),
    MOVE(8, REG(xmm6), TEMP(temp17, 8, 8, FP)),
    MOVE(8, REG(xmm7), TEMP(temp18, 8, 8, FP)),
    STK_LOAD(4, TEMP(temp19, 4, 4, FP), OFFSET(16)),
    JUMP(0, LABEL(.L20)),
  ),
  BLOCK(20,
    RETURN(0),
  ),
)
